examples/http-proxy/client/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/logger"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+)
+
+func main() {
+	if os.Getenv("HTTPS_PROXY") != "https://localhost:8443" && os.Getenv("HTTP_PROXY") != "http://localhost:8443" {
+		log.Printf(`To run this example, first start the proxy server in the examples/http-proxy/proxy directory:
+
+$ go run ../proxy [http|https]
+
+Then, run this example setting the HTTP_PROXY or HTTPS_PROXY environment variable to the proxy server:
+
+$ HTTPS_PROXY=https://localhost:8443 go run .
+or
+$ HTTP_PROXY=http://localhost:8443 go run .
+depending on the proxy type you started.
+`)
+		os.Exit(1)
+	}
+	log.Printf("Constructing client...")
+	logger.DefaultLogger = &logger.SimpleLogger{
+		Level: logger.LevelDebug,
+	}
+	var tlsNextProto map[string]func(authority string, c *tls.Conn) http.RoundTripper
+	if os.Getenv("HTTPS_PROXY") != "" {
```

Comment:
Why check if we already check that a proxy is configured earlier in the function?

Reply:
This is specifically only needed for HTTPS proxies, in which case the client is speaking HTTP/2 directly to the proxy. Go's HTTP client doesn't currently implement sending the CONNECT method correctly over HTTP/2.

The example demonstrates HTTP proxies as well. With an HTTP proxy, Go's HTTP client uses HTTP's CONNECT method over HTTP/1.1 to establish a TLS connection to a remote endpoint. That is where HTTP/2 is negotiated, and that is supported just fine.

Reply:
Actually, I need to take this back. Stupid of me... you obviously can't proxy an HTTPS connection over HTTP, that would be a huge security problem. At least, you can't use HTTP_PROXY when the target endpoint is HTTPS...